lib/utils/combineJSON.js

```diff
@@ -99,19 +99,40 @@ export default async function combineJSON(
 
       // If there is no file_content then no custom parser ran on that file
       if (!file_content) {
-        if (['.js', '.mjs'].includes(extname(filePath))) {
-          let resolvedPath = resolve(filePath, vol?.__custom_fs__);
-          // eslint-disable-next-line no-undef
-          if (isNode && process?.platform === 'win32') {
-            // Windows FS compatibility. If in browser, we use an FS shim which doesn't require this Windows workaround
-            resolvedPath = new URL(`file:///${resolvedPath}`).href;
+        switch (extname(filePath)) {
+          case '.js':
+          case '.mjs':
+          case '.ts': {
+            let resolvedPath = resolve(filePath, vol?.__custom_fs__);
+            // eslint-disable-next-line no-undef
+            if (isNode && process?.platform === 'win32') {
+              // Windows FS compatibility. If in browser, we use an FS shim which doesn't require this Windows workaround
+              resolvedPath = new URL(`file:///${resolvedPath}`).href;
+            }
+            file_content = (await import(/* @vite-ignore */ /* webpackIgnore: true */ resolvedPath))
```

Comment:
Just a soft suggestion, perhaps it makes sense to warn the user that they need the experimental flag for NodeJS (or use Bun/Deno) if the import of .ts file fails.

Reply:
Yeah had similar thoughts. I'll add that.